### PR TITLE
SC - 5780 - teacher edits team

### PR DESCRIPTION
**Pull request by Grzesiek80**
# Description
create new test

## Links to Tickets or other pull requests

https://ticketsystem.hpi-schul-cloud.org/browse/SC-5780

### CODE REVIEW

File: features/teams/createTeam.feature [hunk 1]
```diff
@@ -7,13 +7,14 @@ Feature: Set of tests to create teams
 	@createTeamWithTwoMembers
 	Scenario Outline: As a user, I want to be able to create a team with two new members
 		Given <userRole> logs in with email '<username>' and password '<password>'
-		When <userRole> creates a new team with name '<teamName>' and description '<description>' and color orange
+		When <userAdmin> creates a new team with name '<teamName>' and description '<description>' and color '<color>'
```
Comment (alstanki) [inline features/teams/createTeam.feature]:
```suggestion
		When <userRole> creates a new team with name '<teamName>' and description '<description>' and color '<color>'
```

File: features/management_students/deleteStudent.feature [hunk 2]
```diff
@@ -27,22 +27,23 @@ Feature: Set of tests to delete students
...
 		And <userAdmin> clicks Delete-user button inside popup
 		Then <userAdmin> goes to Teams Page
 		When <userAdmin> clicks on Member icon in team with name '<teamName>'
-		Then should see that team members: '<firstName> <lastName>' are not listed
+		Then <userAdmin> should see that team members: '<firstName> <lastName>' are not listed
 
 		Examples:
-			| userAdmin | adminUsername                | adminPassword  | studentUserName 				  | firstName | lastName | teamName  | description      | orange  | number |
-			| admin     | kai.admin.qa@schul-cloud.org | Schulcloud1qa! | boris.wasser.qa@schul-cloud.org | Boris	  | Wasser   | test team | test description | #ffad42 | 2      |
+			| userAdmin | adminUsername                | adminPassword  | studentUserName 				  | firstName | lastName | teamName  | description      | color  | number |
```
Comment (alstanki) [inline features/management_students/deleteStudent.feature:45]: Why userAdmin?

File: features/teams/createTeam.feature [hunk 3]
```diff
@@ -7,13 +7,14 @@ Feature: Set of tests to create teams
...
 		And <userRole> clicks Submit-add-team-member button
-		Then <userRole> sees that team with name '<teamName>', colour '<colour>' and  member number '<number>' is visible on the list
+		Then <userRole> goes to Teams Page
+		Then <userRole> sees that team with name '<teamName>', colour '<color>' and  member number '<number>' is visible on the list
 		When <userRole> clicks on Member icon in team with name '<teamName>'
 		Then <userRole> should see that team members: '<teacherName>,<firstName1> <lastName1>,<firstName2> <lastName2>' are listed
 		Examples:
-			| userRole | username               		 | password       | teamName  | description      | teacherName | firstName1 | lastName1 | firstName2 | lastName2  | colour  | number |
-			| teacher  | karl.teacher.qa@schul-cloud.org | Schulcloud1qa! | test team | test description | Karl Herzog | Boris      | Wasser    | Herbert    | Kraft      | #ffad42 | 3      |
+			| userRole | username               		 | password       | teamName  | description      | teacherName | firstName1 | lastName1 | firstName2 | lastName2  | color   | number |
```
Comment (alstanki) [inline features/teams/createTeam.feature]:
```suggestion
			| userRole | userName               		 | password       | teamName  | description      | teacherName | firstName1 | lastName1 | firstName2 | lastName2  | color   | numberOfMembers |
```

File: features/teams/editTeam.feature [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        Given <userRole> logs in with email '<userName>' and password '<password>'
+        Then <userRole> goes to Teams Page
+        And <userRole> sees that team with name '<teamName>' is visible on the list
+        When <userRole> chooses team with name '<teamName>'
+        And <userRole> clicks on Edit-team button
+        And <userRole> changes name of Team '<changeName>'
+        And <userRole> changes team description '<description>'
+        And <userRole> chooses team colour '<color>'
+        And <userRole> clicks on Save-changes in team button
+        Then <userRole> goes to Teams Page
```
Comment (alstanki) [inline features/teams/editTeam.feature:18]:
```suggestion
        And <userRole> goes to Teams Page
```

File: features/teams/editTeam.feature [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+        And <userRole> sees that team with name '<changeName>' is visible on the list
```
Comment (alstanki) [inline features/teams/editTeam.feature:19]:
```suggestion
        Then <userRole> sees that team with name '<changeName>' is visible on the list
```

File: page-objects/pages/DashboardPage.js [hunk 6]
```diff
@@ -13,8 +13,8 @@ const taskElement = {
 };
 
 const taskSection = {
-	assigned: '#main-content > section:nth-child(3)',
-	private: '#main-content > section:nth-child(4)',
+	assigned: 'section[data-testid="published-tasks-section"]',
```
Comment (alstanki) [inline page-objects/pages/DashboardPage.js:16]:
```suggestion
	published: 'section[data-testid="published-tasks-section"]',
```
one naming would be better :)

File: page-objects/pages/teamsPages/TMSAddEditTeamPage.js [hunk 7]
```diff
@@ -9,7 +9,37 @@ const elementHelpers = require('../../../runtime/helpers/elementHelpers.js');
...
+			console.error(`This color: ${color} does not exist on the list of possible choices`);
+			break;
+		
+	}
+	return colorItem;
+}
+
 
 // submit button
 const createTeamBtn = 'button[data-testid="create_team_btn"]';
```
Comment (alstanki) [inline page-objects/pages/teamsPages/TMSAddEditTeamPage.js:45]: Paste this please on top of the file where is the rest of the selectors.

File: page-objects/pages/teamsPages/TMSAddEditTeamPage.js [hunk 8]
```diff
@@ -22,27 +52,28 @@ async function setTeamDescription(description) {
 	await waitHelpers.waitAndSetValue(teamDescription, description);
 }
 
-async function selectColor() {
+async function selectColor(color) {
```
Comment (alstanki) [inline page-objects/pages/teamsPages/TMSAddEditTeamPage.js]: I know it's not your implementation :) but it can be done by the way.  setColor

**Review by d3n153 — APPROVED** [review]
Fine from my side.

File: page-objects/pages/teamsPages/TMSTeamListPage.js [hunk 9]
```diff
@@ -131,14 +161,40 @@ async function clickMemberIconInTeam(teamName) {
...
+		? `${defaultString} should be visible on the list`
+		: `${defaultString} should not be visible on the list`;
+
+	const resultMsg = 'Actual list of teams: ' + (await getListOfTeamTitles());
+	const isTeam = await isTeamOnList(teamName);
+	expect(isTeam, msg + resultMsg).to.equal(expectedValue);
+
+}
+
+async function isTeamOnList(teamName) {
```
Comment (alstanki) [inline page-objects/pages/teamsPages/TMSTeamListPage.js:183]: Separate method not needed. Use everywhere isTeamVisible

File: features/teams/createTeam.feature (hunk 1 above)
Comment (Grzesiek80) [reply features/teams/createTeam.feature]: done

File: features/management_students/deleteStudent.feature (hunk 2 above)
Comment (Grzesiek80) [reply features/management_students/deleteStudent.feature:45]: done

File: features/teams/createTeam.feature (hunk 3 above)
Comment (Grzesiek80) [reply features/teams/createTeam.feature]: done

File: features/teams/editTeam.feature (hunk 4 above)
Comment (Grzesiek80) [reply features/teams/editTeam.feature:18]: done

File: features/teams/editTeam.feature (hunk 5 above)
Comment (Grzesiek80) [reply features/teams/editTeam.feature:19]: done

File: page-objects/pages/DashboardPage.js (hunk 6 above)
Comment (Grzesiek80) [reply page-objects/pages/DashboardPage.js:16]: done

File: page-objects/pages/teamsPages/TMSAddEditTeamPage.js (hunk 7 above)
Comment (Grzesiek80) [reply page-objects/pages/teamsPages/TMSAddEditTeamPage.js:45]: done

File: page-objects/pages/teamsPages/TMSAddEditTeamPage.js (hunk 8 above)
Comment (Grzesiek80) [reply page-objects/pages/teamsPages/TMSAddEditTeamPage.js]: done

File: page-objects/pages/teamsPages/TMSTeamListPage.js (hunk 9 above)
Comment (Grzesiek80) [reply page-objects/pages/teamsPages/TMSTeamListPage.js:183]: i removed another one this method is used in isTeamVisible

File: step_definitions/teams/createTeam_steps.js [hunk 10]
```diff
@@ -5,8 +5,8 @@ const TMSTeamListPage = require('../../page-objects/pages/teamsPages/TMSTeamList
 const TMSTeamMembersPage = require('../../page-objects/pages/teamsPages/TMSTeamMembersPage.js');
 
 //WHEN
-When(/^.*creates a new team with name '([^']*)' and description '([^']*)' and color orange$/, function (teamName, description) {
-	return TMSAddEditTeamPage.createTeamAndGoToInternalMembersManagement(teamName, description);
+When(/^.*creates a new team with name '([^']*)' and description '([^']*)' and color '([^']*)'$/, function (teamName, description, colour) {
+	return TMSAddEditTeamPage.createTeamAndGoToInternalMembersManagement(teamName, description, colour);
```
Comment (alstanki) [inline step_definitions/teams/createTeam_steps.js:9]:
```suggestion
	await TMSAddEditTeamPage.createTeamAndGoToInternalMembersManagement(teamName, description, colour);
```
The same in other similar places

**Review by alstanki — CHANGES_REQUESTED** [review]

**Review by alstanki — APPROVED** [review]

**Pull request merged**